sparkmagic/sparkmagic/tests/test_sparkstorecommand.py

```diff
@@ -0,0 +1,241 @@
+    code = "hugedataset.collect()"
+    sparkcommand = SparkStoreCommand(code, variable_name, samplemethod='take', maxrows=100)
+    assert_equals(sparkcommand._scala_command(variable_name),
+                  Command('{}.toJSON.take(100).foreach(println)'.format(variable_name)))
+
+    sparkcommand = SparkStoreCommand(code, variable_name, samplemethod='take', maxrows=-1)
+    assert_equals(sparkcommand._scala_command(variable_name),
+                  Command('{}.toJSON.collect.foreach(println)'.format(variable_name)))
+
+    sparkcommand = SparkStoreCommand(code, variable_name, samplemethod='sample', samplefraction=0.25, maxrows=-1)
+    assert_equals(sparkcommand._scala_command(variable_name),
+                  Command('{}.toJSON.sample(false, 0.25).collect.foreach(println)'.format(variable_name)))
+
+    sparkcommand = SparkStoreCommand(code, variable_name, samplemethod='sample', samplefraction=0.33, maxrows=3234)
+    assert_equals(sparkcommand._scala_command(variable_name),
+                  Command('{}.toJSON.sample(false, 0.33).take(3234).foreach(println)'.format(variable_name)))
+
+
+
+@with_setup(_setup, _teardown)
+def test_r_livy_sampling_options():
+    variable_name = "abc"
+    code = "hugedataset.collect()"
+    sparkcommand = SparkStoreCommand(code, variable_name, samplemethod='take', maxrows=100)
+
+    assert_equals(sparkcommand._r_command(variable_name),
+                  Command('for ({} in (jsonlite::toJSON(take({},100)))) {{cat({})}}'.format(LONG_RANDOM_VARIABLE_NAME, variable_name, LONG_RANDOM_VARIABLE_NAME)))
+
+    sparkcommand = SparkStoreCommand(code, variable_name, samplemethod='take', maxrows=-1)
+    assert_equals(sparkcommand._r_command(variable_name),
+                  Command('for ({} in (jsonlite::toJSON(collect({})))) {{cat({})}}'.format(LONG_RANDOM_VARIABLE_NAME, variable_name, LONG_RANDOM_VARIABLE_NAME)))
+
+    sparkcommand = SparkStoreCommand(code, variable_name, samplemethod='sample', samplefraction=0.25, maxrows=-1)
+    assert_equals(sparkcommand._r_command(variable_name),
+                  Command('for ({} in (jsonlite::toJSON(collect(sample({}, FALSE, 0.25))))) {{cat({})}}'.format(LONG_RANDOM_VARIABLE_NAME, variable_name, LONG_RANDOM_VARIABLE_NAME)))
+
+    sparkcommand = SparkStoreCommand(code, variable_name, samplemethod='sample', samplefraction=0.33, maxrows=3234)
+    assert_equals(sparkcommand._r_command(variable_name),
+                 Command('for ({} in (jsonlite::toJSON(take(sample({}, FALSE, 0.33),3234)))) {{cat({})}}'.format(LONG_RANDOM_VARIABLE_NAME, variable_name, LONG_RANDOM_VARIABLE_NAME)))
+
+
+@with_setup(_setup, _teardown)
+def test_execute_code():
+    spark_events = MagicMock()
+    variable_name = "abc"
+    code = "hugedataset.collect()"
+    sparkcommand = SparkStoreCommand(code, variable_name, "take", 100, 0.2, spark_events=spark_events)
+    sparkcommand.to_command = MagicMock(return_value=MagicMock())
+    result = """{"z":100, "nullv":null, "y":50}
```

Comment:
Do we have unit tests for the code that transforms the strings to dataframes? I think we do... if so, we should move those tests to the utils tests when you move the function over. That way, you don't have to test that here, and you need only mock the utils function.